### ISSUE TITLE
Exporting to mysql error: Packet for query is too large

### ISSUE THREAD

**Issue by Laurira** [keeps/db-preservation-toolkit#251]
Description: 

Bruno: "I am sending multiple INSERT queries at once to MySQL (like 100 or so at a time).
This problem happens when the length of those queries combined exceeds 4KB (4194304 Bytes). And those big queries are not run on the server, so the data is in the SIARD but does not get converted to MySQL. So I will need to check this limit and avoid reaching it somehow…".

At the moment this is not a pilot blocker as the person who is dealing with the pilot is on vacation and we have not reached to the point to know if this bug occurs in the pilot also.

```
2016-08-10 10:38:05,326 [main] ERROR (c.d.m.j.o.JDBCExportModule) Error executing part of a batch of queries
java.sql.BatchUpdateException: Packet for query is too large (4528379 > 4194304). You can change this value on the server by setting the max_allowed_packet' variable.
                at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
                at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
                at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
                at java.lang.reflect.Constructor.newInstance(Unknown Source)
                at com.mysql.jdbc.Util.handleNewInstance(Util.java:404)
                at com.mysql.jdbc.Util.getInstance(Util.java:387)
                at com.mysql.jdbc.SQLError.createBatchUpdateException(SQLError.java:1154)
                at com.mysql.jdbc.PreparedStatement.executeBatchedInserts(PreparedStatement.java:1582)
                at com.mysql.jdbc.PreparedStatement.executeBatchInternal(PreparedStatement.java:1248)
                at com.mysql.jdbc.StatementImpl.executeBatch(StatementImpl.java:959)
                at com.databasepreservation.modules.jdbc.out.JDBCExportModule.handleDataRow(JDBCExportModule.java:480)
                at com.databasepreservation.modules.siard.in.content.SIARD2ContentImportStrategy.endElement(SIARD2ContentImportStrategy.java:296)
                at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.endElement(Unknown Source)
                at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.endElement(Unknown Source)
                at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanEndElement(Unknown Source)
                at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(Unknown Source)
                at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
                at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(Unknown Source)
                at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
                at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
                at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
                at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
                at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
                at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
                at com.databasepreservation.modules.siard.in.content.SIARD2ContentImportStrategy.importContent(SIARD2ContentImportStrategy.java:141)
                at com.databasepreservation.modules.siard.in.input.SIARDImportDefault.getDatabase(SIARDImportDefault.java:45)
                at com.databasepreservation.Main.run(Main.java:126)
                at com.databasepreservation.Main.internal_main(Main.java:80)
                at com.databasepreservation.Main.main(Main.java:62)
Caused by: com.mysql.jdbc.PacketTooBigException: Packet for query is too large (4528379 > 4194304). You can change this value on the server by setting the max_allowed_packet' variable.
                at com.mysql.jdbc.MysqlIO.send(MysqlIO.java:3583)
                at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2460)
                at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2625)
                at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2551)
                at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1861)
                at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2073)
                at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2009)
                at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5094)
                at com.mysql.jdbc.PreparedStatement.executeBatchedInserts(PreparedStatement.java:1528)
                ... 21 common frames omitted
2016-08-10 10:38:05,339 [main] DEBUG (c.d.m.Reporter) something failed, message: - Information: In table `public.tbl_access_application`, inserting rows with index from 808 to 908  failed because  there was an error with at least one of the rows
2016-08-10 10:38:05,795 [main] DEBUG (c.d.m.s.i.c.SIARD2ContentImportStrategy) BLOB cell public.tbl_access_application.application_signed_doc_content.923 on row #923 with lob dir record924.bin
```

Attach the dbptk-app.log.txt file below.
[vauExport.txt](https://github.com/keeps/db-preservation-toolkit/files/414803/vauExport.txt)

**Comment by chalkos** [comment]
Maybe this can be fixed by obtaining this limit and avoid reaching it (and warn if it is too low).

**Comment by Laurira** [comment]
Can I set this limit via command line or you have to fix it in the code?

**Comment by chalkos** [comment]
To fix the issue as a whole, I must change the code.

But as a temporary fix you can try to follow these steps:
http://stackoverflow.com/a/8062538/1483200 to set a larger limit for the
allowed query packet size
And you can use this
http://dev.mysql.com/doc/refman/5.7/en/option-files.html to find the
options file.

**Comment by chalkos** [comment]
The fix worked on both MySQL 5.6 and MariaDB 10.3
